lib/compat/wordpress-6.5/scripts-modules.php

```diff
@@ -0,0 +1,304 @@
+ *                                                                                 or path of the module relative to
+ *                                                                                 the WordPress root directory. If
+ *                                                                                 it is provided and the module has
+ *                                                                                 not been registered yet, it will be
+ *                                                                                 registered.
+ * @param array<string|array{id: string, import?: 'static'|'dynamic' }> $deps      Optional. An array of module
+ *                                                                                 identifiers of the dependencies of
+ *                                                                                 this module. The dependencies can
+ *                                                                                 be strings or arrays. If they are
+ *                                                                                 arrays, they need an `id` key with
+ *                                                                                 the module identifier, and can
+ *                                                                                 contain an `import` key with either
+ *                                                                                 `static` or `dynamic`. By default,
+ *                                                                                 dependencies that don't contain an
+ *                                                                                 `import` key are considered static.
+ * @param string|false|null                                             $version   Optional. String specifying the
+ *                                                                                 module version number. Defaults to
+ *                                                                                 false. It is added to the URL as a
+ *                                                                                 query string for cache busting
+ *                                                                                 purposes. If $version is set to
+ *                                                                                 false, the version number is the
+ *                                                                                 currently installed WordPress
+ *                                                                                 version. If $version is set to
+ *                                                                                 null, no version is added.
+ */
+function gutenberg_enqueue_module( $module_id, $src = '', $deps = array(), $version = false ) {
```

Comment:
We also need the `wp_enqueue_module` function names.

Reply:
It's possible that we should not have the `gutenberg_enqueue_module` functions in this file, so it can be copy-pasted with new changes.

Reply:
I renamed the functions with the ones with `wp_` with a declaration prevention.
Should we move them to another file then?

I guess changes should be done first in this file, and later in `wordpress-develop`. I think is good to have them in the same file than Core.